Modules/TopGear/TopGear.lua

```diff
@@ -183,20 +183,31 @@ function CraftSim.TOPGEAR:GetProfessionGearFromInventory(recipeData)
 
         for bag = BANK_CONTAINER, NUM_BAG_SLOTS + NUM_BANKBAGSLOTS do
             for slot = 1, C_Container.GetContainerNumSlots(bag) do
-                local itemLink = C_Container.GetContainerItemLink(bag, slot)
-                if itemLink ~= nil then
-                    local itemSubType = select(3, C_Item.GetItemInfoInstant(itemLink))
-                    if itemSubType == currentProfession then
-                        local professionGear = CraftSim.ProfessionGear()
-                        professionGear:SetItem(itemLink)
-                        table.insert(inventoryGear, professionGear)
-
-                        if not GUTIL:Find(CraftSimRecipeDataCache.professionGearCache[crafterUID]
-                                [recipeData.professionData.professionInfo.profession]
-                                .availableProfessionGear, function(g) return g.itemLink == itemLink end) then
-                            table.insert(CraftSimRecipeDataCache.professionGearCache[crafterUID]
-                                [recipeData.professionData.professionInfo.profession]
-                                .availableProfessionGear, professionGear:Serialize())
+                local itemLoc = ItemLocation:CreateFromBagAndSlot(bag,slot)
+                if itemLoc:IsValid() then
+                    local isCached = C_Item.IsItemDataCached(itemLoc)
+                    if not isCached then
+                        C_Item.RequestLoadItemData(itemLoc)
+                    end
+                    
+                    local itemLink = C_Item.GetItemLink(ItemLocation:CreateFromBagAndSlot(bag,slot))
+                    if itemLink ~= nil then
+                        local itemSubType = select(3, C_Item.GetItemInfoInstant(itemLink))
+                        local itemEquipLoc = select(4, C_Item.GetItemInfoInstant(itemLink))
+                        if itemSubType == currentProfession and (itemEquipLoc == "INVTYPE_PROFESSION_TOOL" or itemEquipLoc == "INVTYPE_PROFESSION_GEAR") then

```

Comment:
still wierded out by your bug that non profession tools are cached cause as you can see here the inventory type is checked against

Reply:
No, the check of inventory type filter correctly.
There were still the old profession tools whose name was not present because the item was not in the cache ^^